### PR TITLE
algorithms task #5 tdobrovolska

### PR DESCRIPTION
**Pull request by TetianaDobrovolska**

### CODE REVIEW

File: algorithms task (5)/student/include/impl.h [hunk 1]
```diff
@@ -10,4 +12,11 @@
  * @return gathered range
  */
 template <typename BiIt, typename UnaryPredicate>
-std::pair<BiIt, BiIt> gather(BiIt begin, BiIt end, BiIt position, UnaryPredicate predicate);
+std::pair<BiIt, BiIt> gather(BiIt begin, BiIt end, BiIt position, UnaryPredicate predicate)
+{
+    using Type = decltype(*begin);
```
Comment (smay1613) [inline algorithms task (5)/student/include/impl.h:17]: _please give more descriptive name or use iterator traits_

File: algorithms task (5)/student/include/impl.h [hunk 2]
```diff
@@ -10,4 +12,11 @@
...
+    auto not_predicate = [predicate](Type& x) { return !predicate(x); };
+    auto first = std::stable_partition(begin, position, not_predicate);
```
Comment (smay1613) [inline algorithms task (5)/student/include/impl.h:19]: _stable partition has overhead_

File: algorithms task (5)/student/include/impl.h (hunk 1 above)
Comment (TetianaDobrovolska) [reply algorithms task (5)/student/include/impl.h]: fixed

File: algorithms task (5)/student/include/impl.h (hunk 2 above)
Comment (TetianaDobrovolska) [reply algorithms task (5)/student/include/impl.h]: fixed

**Pull request closed without merging**